docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,277 @@
+# Firecracker Snapshotting
+
+## What is microVM snapshotting?
+
+MicroVM snapshotting is a mechanism through which a running microVM, including
+the guest environment, can be serialized and saved to an external medium in the
+form of a `snapshot`. This snapshot can be later used to restore a microVM
+with its guest workload at that particular point in time.
+
+## Snapshotting in Firecracker
+
+Firecracker snapshotting allows the creation of a microVM snapshot. This
+snapshot can be later loaded in a possibly different Firecracker process and
+the original guest workload is simply resumed. It is not guaranteed though that
+the state of the network connections survives the process.
+
+In order to make restoring possible, Firecracker snapshots save the full state
+of the following resources:
+- the guest memory,
+- the emulated HW state (both KVM and Firecracker emulated HW),
+- the guest's storage disks' contents.
```

Comment:
Firecracker doesn't actively do anything to save the disk separately. That's called out a few lines below; I'd remove mention of disk contents here.